File: client/src/app/modules/quotation/services/provision.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { ProvisionType } from '../../quotation/model/ProvisionType';

@Injectable({
  providedIn: 'root'
})
export class ProvisionTypeService extends AppRestService<ProvisionType>{

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getProvisionTypes() {
    return this.getListCached(new HttpParams(), "provision-types");
  }

  addOrUpdateProvisionType(provisionType: ProvisionType) {
    this.clearListCache(new HttpParams(), "provision-types");
    return this.addOrUpdate(new HttpParams(), "provision-type", provisionType, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/quotation/service/ProvisionTypeService.java
package com.jss.osiris.modules.osiris.quotation.service;

import java.util.List;

import com.jss.osiris.modules.osiris.quotation.model.ProvisionType;

public interface ProvisionTypeService {
    public List<ProvisionType> getProvisionTypes();

    public ProvisionType getProvisionType(Integer id);

    public ProvisionType addOrUpdateProvisionType(ProvisionType provisionType);
}
